neo/Consensus/ConsensusService.cs

```diff
@@ -403,14 +745,26 @@ public static Props Props(IActorRef localNode, IActorRef taskManager, Store stor
             return Akka.Actor.Props.Create(() => new ConsensusService(localNode, taskManager, store, wallet)).WithMailbox("consensus-service-mailbox");
         }
 
+        private byte GetLastExpectedView(int validatorIndex)
+        {
+            var lastPreparationPayload = context.PreparationPayloads[validatorIndex];
+            if (lastPreparationPayload != null)
+                return lastPreparationPayload.GetDeserializedMessage<ConsensusMessage>().ViewNumber;
+
+            return context.ChangeViewPayloads[validatorIndex]?.GetDeserializedMessage<ChangeView>().NewViewNumber ?? (byte) 0;
+        }
+
         private void RequestChangeView()
         {
             context.State |= ConsensusState.ViewChanging;
-            context.ExpectedView[context.MyIndex]++;
-            Log($"request change view: height={context.BlockIndex} view={context.ViewNumber} nv={context.ExpectedView[context.MyIndex]} state={context.State}");
-            ChangeTimer(TimeSpan.FromSeconds(Blockchain.SecondsPerBlock << (context.ExpectedView[context.MyIndex] + 1)));
-            localNode.Tell(new LocalNode.SendDirectly { Inventory = context.MakeChangeView() });
-            CheckExpectedView(context.ExpectedView[context.MyIndex]);
+            byte expectedView = GetLastExpectedView(context.MyIndex);
+            expectedView++;
+            Log($"request change view: height={context.BlockIndex} view={context.ViewNumber} nv={expectedView} state={context.State}");
+            ChangeTimer(TimeSpan.FromSeconds(Blockchain.SecondsPerBlock << (expectedView + 1)));
```

Comment:
We should add a max time here (inside `ChangeTimer`)

Reply:
Why would it matter?